File: leehyowonzero/12week/jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        MaxCanjump = 0
        for i in range(len(nums)):
            if(i <= MaxCanjump):
                MaxCanjump = max(MaxCanjump,i + nums[i])
            else:
                return False
            
            if(MaxCanjump >= len(nums) - 1):
                return True

File: JeongGod/12week/leetcode/JumpGame.java
import java.util.*;

class JumpGame {
    public boolean canJump(int[] nums) {
        int maxIdx = 0;
        if (nums.length == 1) return true;
        Deque<List<Integer>> dq = new ArrayDeque<>();
        dq.addLast(Arrays.asList(0, nums[0]));

        while (dq.size() > 0) {
            List<Integer> result = dq.removeFirst();
            int cur = result.get(0);
            int maxJump = result.get(1);

            for (int i=maxIdx+1; i<=cur+maxJump; i++) {
                if (i >= nums.length-1) {
                    return true;
                }

                dq.addLast(Arrays.asList(i, nums[i]));
            }

            maxIdx = Math.max(maxIdx, cur+maxJump);
        }
        return false;
    }
}